app/views/job_invocations/_tab_hosts.html.erb

```diff
@@ -10,24 +10,24 @@
     </thead>
 
     <tbody>
-      <% job_invocation.last_task.sub_tasks.for_action_types('Actions::RemoteExecution::RunHostJob').each do |task| %>
+      <% job_invocation.targeting.hosts.each do |host| %>
         <tr>
-          <% template_invocation = task.locks.where(:resource_type => 'TemplateInvocation').first.try(:resource) %>
-          <% if (host = task.locks.where(:resource_type => 'Host::Managed').first.try(:resource)) %>
-            <td><%= link_to_if_authorized host.name, hash_for_host_path(host).merge(:auth_object => host, :permission => :view_hosts) %></td>
-          <% else %>
-            <td><%= _('host not linked') %></td>
-          <% end %>
+          <% task = job_invocation.last_task.sub_tasks.select { |task| task.locks.where(:resource_type => 'Host::Managed', :resource_id => host.id).any? }.first %>
+          <% template_invocation = task.locks.where(:resource_type => 'TemplateInvocation').first.try(:resource) unless task.nil? %>
+          <td><%= link_to_if_authorized host.name, hash_for_host_path(host).merge(:auth_object => host, :permission => :view_hosts) %></td>
 
           <td><%= template_invocation.nil? ? _('N/A') : _(RemoteExecutionProvider.provider_for(template_invocation.template.provider_type)) %></td>
```

Comment:
I guess we could extract this into `remote_execution_provider_for_task(task)` helper, the `template_invocation` itself is not used anywhere else